app/models/concerns/profile_tailoring.rb

```diff
@@ -12,6 +12,15 @@ def tailored_rule_ref_ids
     end).to_h
   end
 
+  def rule_group_ancestor_ref_ids
+    parent_groups = RuleGroup.where(id: added_rules.map(&:rule_group_id))
+
+    # Retrieve all unique ancestor rule_group ids including parent group ids
+    path_ids = parent_groups.map(&:path_ids).flatten.uniq
```

Comment:
I don't understand the point of the first query :thinking: the second one is fine, with the last one the issue is that you have to list all the IDs instead of a subquery.

Reply:
This worked and gave me that query that you have in the above comment: 
```
base = RuleGroup.where(id: rules.map(&:rule_group_id))
base.or(
      RuleGroup.where(id: base.select("unnest(string_to_array(ancestry, '/'))::uuid"))
).pluck(:ref_id)
```
Trying to figure out how to rewrite it using arel now

Reply:
You just need the part inside the `select` in arel, probably with `.to_sql`